src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statefulClones = [];
+  const stateClone = { ...state };
+
+  for (const singleAction of actions) {
+    switch (singleAction.type) {
+      case 'addProperties':
+        Object.assign(stateClone, singleAction.extraData);
+        break;
+
+      case 'removeProperties':
+        singleAction.keysToRemove.forEach(el => {
+          delete stateClone[el];
+        });
+        break;
+
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+    };
+
```

Comment:
_[CODE STYLE]: switch/case should always have default case for error handling._

- [ ] add default case for error handling